gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/LocaleHandler.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.gateway.handler.root.resources.handler;
+
+import io.gravitee.am.gateway.handler.vertx.view.thymeleaf.GraviteeMessageResolver;
+import io.vertx.core.Handler;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+import java.util.Locale;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class LocaleHandler implements Handler<RoutingContext> {
+
+    private GraviteeMessageResolver messageResolver;
+
+    public LocaleHandler(GraviteeMessageResolver messageResolver) {
+        this.messageResolver = messageResolver;
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        for(var language : routingContext.acceptableLanguages()) {
+            if (messageResolver.isSupported(new Locale(language.value()), false)) {
+                routingContext.put("lang", language.value());
+                routingContext.next();
+                return;
+            }
+        }
+
+        for(var language : routingContext.acceptableLanguages()) {
+            if (messageResolver.isSupported(new Locale(language.tag()), true)) {
+                routingContext.put("lang", language.tag());
+                routingContext.next();
+                return;
+            }
+        }
```

Comment:
Why we need a double loop ? 

Why not using the routingContext.preferredLanguage() method ?

Reply:
The Accept-Languages header value may contain multiple values with different priorities.
The preferredLanguage returns only the one with highest priority but we do not have guaranty that this preferredLanguage is managed by our dictionnary, so I go through the acceptableLanguages (that are shorted by priority) to get the most accurante language.

About the double loop, its purpose is to manage the language with country code (fr-FR or en-GB). 
for ex, if the Accept-Language contains `fr-FR` but not `fr` and if on AM the admin user only define `fr` as language, we will not be able to find a matching translation. As we do not have guaranty that the value 'fr-FR' will have the same priority as 'fr', I prefere doing a strict matching first and then fallback to a 'lax' matching